components/org.wso2.carbon.identity.tenant.resource.manager/src/main/java/org/wso2/carbon/identity/tenant/resource/manager/TenantAwareAxis2ConfigurationContextObserver.java

```diff
@@ -0,0 +1,188 @@
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.tenant.resource.manager;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.event.publisher.core.config.EventPublisherConfiguration;
+import org.wso2.carbon.event.publisher.core.exception.EventPublisherConfigurationException;
+import org.wso2.carbon.event.stream.core.EventStreamConfiguration;
+import org.wso2.carbon.event.stream.core.exception.EventStreamConfigurationException;
+import org.wso2.carbon.identity.configuration.mgt.core.constant.ConfigurationConstants;
+import org.wso2.carbon.identity.configuration.mgt.core.exception.ConfigurationManagementException;
+import org.wso2.carbon.identity.configuration.mgt.core.model.ResourceFile;
+import org.wso2.carbon.identity.tenant.resource.manager.constants.TenantResourceConstants;
+import org.wso2.carbon.identity.tenant.resource.manager.exception.TenantResourceManagementException;
+import org.wso2.carbon.identity.tenant.resource.manager.internal.TenantResourceManagerDataHolder;
+import org.wso2.carbon.utils.AbstractAxis2ConfigurationContextObserver;
+import org.wso2.carbon.utils.multitenancy.MultitenantConstants;
+import java.util.List;
+
+import static org.wso2.carbon.identity.tenant.resource.manager.constants.TenantResourceConstants.ErrorMessages.ERROR_CODE_ERROR_WHEN_CREATING_TENANT_EVENT_PUBLISHER_CONFIGURATION_BY_CONFIG_STORE;
+import static org.wso2.carbon.identity.tenant.resource.manager.constants.TenantResourceConstants.ErrorMessages.ERROR_CODE_ERROR_WHEN_CREATING_TENANT_EVENT_PUBLISHER_CONFIGURATION_USING_SUPER_TENANT_CONFIG;
+import static org.wso2.carbon.identity.tenant.resource.manager.util.ResourceUtils.getResourceFile;
+import static org.wso2.carbon.identity.tenant.resource.manager.util.ResourceUtils.getResourceManager;
+import static org.wso2.carbon.identity.tenant.resource.manager.util.ResourceUtils.handleServerException;
+
+/**
+ * Axis2Observer for generating tenant wise publisher configurations.
+ */
+public class TenantAwareAxis2ConfigurationContextObserver extends AbstractAxis2ConfigurationContextObserver {
+
+    private static final Log log = LogFactory.getLog(TenantAwareAxis2ConfigurationContextObserver.class);
+
+    /**
+     * Add the tenant wise publisher and stream Configuration in tenant loading.
+     *
+     * @param tenantId tenant ID.
+     */
+    @Override
+    public void creatingConfigurationContext(int tenantId) {
+
+        log.info("creating configuration context for tenant id: " + tenantId);
+        String tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        List<EventPublisherConfiguration> activeEventPublisherConfigurations;
+        List<EventStreamConfiguration> eventStreamConfigurationList;
+
+        startSuperTenantFlow();
+
+        activeEventPublisherConfigurations = getSuperTenantEventPublisherConfigurations(tenantId);
```

Comment:
You have started a super tenant flow, yet have passed the tenantId passed in input.
What's the intention here

Reply:
Ok noted that you have passed that for logging purposes. But the method definition seems confusing. Maybe you can fix it accordingly

Reply:
done